File: Bridge/KovalchukD-03/ConsoleApp25/rectangle.cs
using System;

namespace BridgePattern
{
    class Rectangle : Shape
    {
        private double sideA;
        private double sideB;

        public Rectangle(double sideA, double sideB, string color) : base(color)
        {
            if (sideA <= 0 || sideB <= 0)
            {
                throw new ArgumentException("Sides must be positive numbers.");
            }

            this.sideA = sideA;
            this.sideB = sideB;
        }

        public override double Area()
        {
            return sideA * sideB;
        }

        public override double Perimeter()
        {
            return 2 * (sideA + sideB);
        }

        public override void Draw()
        {
            Console.WriteLine($"Rectangle: Perimeter={Perimeter()}, Area={Area()}, Color={color}");
        }
    }
}

File: Bridge/Vasylyshyn/src/Shape/Rectangle.java
package Shape;

import Color.Color;

public class Rectangle extends Shape {
    public int SideB;
    public int SideA;
    public Rectangle() {
    }

    public Rectangle(int sideB, int sideA) {
        SideB = sideB;
        SideA = sideA;
    }

    public Rectangle(Color color, int sideB, int sideA) {
        super(color);
        SideB = sideB;
        SideA = sideA;
    }

    @Override
    public String toString() {
        return "Rectangle{" +
                "SideA=" + SideA +
                ", SideB=" + SideB +
                ", color=" + color +
                '}';
    }

    @Override
    public void draw() {
        System.out.print("It is Rectangle. It is already painted ");
        color.Paint();
    }
    @Override
    public double Area(){
        return this.SideA*this.SideB;
    }

    @Override
    public double Perimeter() {
        return (this.SideA+this.SideB)*2;
    }


    public int getSideB() {
        return SideB;
    }

    public Rectangle setSideB(int sideB) {
        SideB = sideB;
        return this;
    }

    public int getSideA() {
        return SideA;
    }

    public Rectangle setSideA(int sideA) {
        SideA = sideA;
        return this;
    }
}
